### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes
Update code to follow modern coding conventions

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
ReSharper'd the code base - LGTM

File: src/Akka.Persistence.Sql.Linq2Db/Journal/Types/WriteQueueSet.cs [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-            TCS = tcs;
+            Tcs = tcs;
             Rows = rows;
         }
 
-        public Seq<JournalRow> Rows { get; set; }
+        public Seq<JournalRow> Rows { get; }
 
-        public List<TaskCompletionSource<NotUsed>> TCS { get; }
+        public ImmutableList<TaskCompletionSource<NotUsed>> Tcs { get; }
```
Comment (to11mtm) [inline src/Akka.Persistence.Sql.Linq2Db/Journal/Types/WriteQueueSet.cs:17]: This was a `List<T>` for perf reasons. Did we bench before and after this change?

I know it's _ugly_ from a purity standpoint but ImmutableList results in a lot more GC pressure and impact on iteration speed.

File: src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs [hunk 2]
```diff
@@ -12,204 +12,191 @@
...
-                        oldRows.Rows = oldRows.Rows.Concat(newRows.Rows);
-                        return oldRows; //.Concat(newRows.Item2).ToList());
-                    })
-                .SelectAsync(_journalConfig.DaoConfig.Parallelism,
-                    async (promisesAndRows) =>
+                .Queue<WriteQueueEntry>(JournalConfig.DaoConfig.BufferSize, OverflowStrategy.DropNew)
+                .BatchWeighted(
+                    JournalConfig.DaoConfig.BatchSize,
+                    cf => cf.Rows.Count,
+                    r => new WriteQueueSet(ImmutableList.Create(new[] {r.Tcs}), r.Rows),
```
Comment (to11mtm) [inline src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs:64]: Per my other comment I wonder if we benchmarked this change against both basic and heavy parallelism cases?

File: src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs [hunk 3]
```diff
@@ -12,204 +12,191 @@
...
 
         private async Task InsertSingle(Seq<JournalRow> xs)
         {
             //If we are writing a single row,
             //we don't need to worry about transactions.
-            using (var db = _connectionFactory.GetConnection())
-            {
-                await db.InsertAsync(xs.Head);
-            }
+            await using var db = ConnectionFactory.GetConnection();
```
Comment (to11mtm) [inline src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs:147]: NIT/Don Quixote opinion:

I'm not a fain of Async dispose on Connections.

In the case of Connections, as long as you have a healthy connection pool you should normally not experience blocking on dispose and the task transitions aren't worth it from observations.

**Pull request merged**